src/ardrive.ts

```diff
@@ -573,7 +593,14 @@ export class ArDrive extends ArDriveAnonymous {
 
 	/** Computes the size of a private file encrypted with AES256-GCM */
 	encryptedDataSize(dataSize: ByteCount): ByteCount {
-		return (dataSize / 16 + 1) * 16;
+		if (dataSize < 0 || !Number.isInteger(dataSize)) {
+			throw new Error(`dataSize must be non-negative, integer value! ${dataSize} is invalid!`);
+		}
+		if (dataSize > Number.MAX_SAFE_INTEGER - 16) {
+			throw new Error(`Max unencrypted dataSize allowed is ${Number.MAX_SAFE_INTEGER - 16}!`);
+		}
+		const modulo16 = dataSize % 16;
+		return dataSize - modulo16 + 16;
```

Comment:
Whats the difference here? And I wonder why it worked with your `file-size` test command code before.